src/create-context.js

```diff
@@ -1,9 +1,18 @@
 import { enqueueRender } from './component';
 
-export let i = 0;
+export let nextContextId = 0;
 
-export function createContext(defaultValue, contextId) {
-	contextId = '__cC' + i++;
+const providers = new Set();
+
+export const unsubscribeFromContext = component => {
+	// if this was a context provider, delete() returns true and we exit:
+	if (providers.delete(component)) return;
```

Comment:
Huh, would've thought calling `Set.delete` on every unmount would've regressed 02_replace1k unmounting perf but it seems fine. Neat!

Reply:
I wonder if we should check this with the synchronous test runner.